src/v/storage/parser.cc

```diff
@@ -312,7 +316,8 @@ class copy_helper {
     ss::future<result<size_t>> run() {
         size_t consumed = 0;
         bool stop = false;
-        while (!stop) {
+        // todo:
+        while (!stop && (*_as).get().abort_requested()) {
```

Comment:
shouldn't this be the inverse condition, IE `!(*_as).get().abort_requested()`? 

Also since this value is optional should we check that it exists before querying it?

Reply:
done